x-pack/plugins/uptime/server/lib/adapters/telemetry/kibana_telemetry_adapter.ts

```diff
@@ -28,22 +28,153 @@ export class KibanaTelemetryAdapter {
   public static initUsageCollector(usageCollector: UsageCollectionSetup) {
     return usageCollector.makeUsageCollector({
       type: 'uptime',
-      fetch: async () => {
+      fetch: async (callCluster: APICaller) => {
+        this.countNoOfUniqueMonitorAndLocations(callCluster);
         const report = this.getReport();
         return { last_24_hours: { hits: { ...report } } };
       },
       isReady: () => true,
     });
   }
 
-  public static countOverview() {
-    const bucket = this.getBucketToIncrement();
-    this.collector[bucket].overview_page += 1;
+  public static countPageView(pageView: PageViewParams) {
+    const bucketId = this.getBucketToIncrement();
+    const bucket = this.collector[bucketId];
+    if (pageView.page === 'Overview') {
+      bucket.overview_page += 1;
+    }
+    if (pageView.page === 'Monitor') {
+      bucket.monitor_page += 1;
+    }
+    if (pageView.page === 'Settings') {
+      bucket.settings_page += 1;
+    }
+    this.updateDateData(pageView, bucket);
+    return bucket;
+  }
+
+  public static updateDateData(
+    { dateStart, dateEnd, autoRefreshEnabled, autorefreshInterval }: PageViewParams,
+    bucket: UptimeTelemetry
+  ) {
+    const prevDateStart = [...bucket.dateRangeStart].pop();
+    if (!prevDateStart || prevDateStart !== dateStart) {
+      bucket.dateRangeStart.push(dateStart);
+      bucket.dateRangeEnd.push(dateEnd);
+    } else {
+      const prevDateEnd = [...bucket.dateRangeEnd].pop();
+      if (!prevDateEnd || prevDateEnd !== dateEnd) {
+        bucket.dateRangeStart.push(dateStart);
+        bucket.dateRangeEnd.push(dateEnd);
+      }
+    }
+
+    const prevAutorefreshInterval = [...bucket.autorefreshInterval].pop();
+    if (!prevAutorefreshInterval || prevAutorefreshInterval !== autorefreshInterval) {
+      bucket.autorefreshInterval.push(autorefreshInterval);
+    }
+    bucket.autoRefreshEnabled = autoRefreshEnabled;
   }
 
-  public static countMonitor() {
+  public static async countNoOfUniqueMonitorAndLocations(callCluster: APICaller) {
+    const params = {
+      index: INDEX_NAMES.HEARTBEAT,
+      body: {
+        query: {
+          bool: {
+            must: [
+              {
+                range: {
+                  '@timestamp': {
+                    gte: 'now-1d/d',
+                    lt: 'now',
+                  },
+                },
+              },
+            ],
+          },
+        },
+        size: 0,
+        aggs: {
+          unique_monitors: {
+            cardinality: {
+              field: 'monitor.id',
+            },
+          },
+          unique_locations: {
+            cardinality: {
+              field: 'observer.geo.name',
+              missing: 'N/A',
```

Comment:
Do we want this to be `'N/A'`? That'll impact the metrics like average length for location name, as well as `min` in most cases, right?

Reply:
no, this will not impact length, it's only used for count. Essentially counting all un-named locations as 1.

Reply:
ah ok, thought it would impact location name. Carry on...